gradio/components/gallery.py

```diff
@@ -176,12 +179,12 @@ def _save(img):
                 img, caption = img
             if isinstance(img, np.ndarray):
                 file = processing_utils.save_img_array_to_cache(
-                    img, cache_dir=self.GRADIO_CACHE
+                    img, cache_dir=self.GRADIO_CACHE, format=self.format or "png"
```

Comment:
If we make the change where format is, by default, png, we won't have to handle the case where it has to be None here and below